service/storage_service.cc

```diff
@@ -1455,6 +1481,9 @@ class topology_coordinator {
             co_return;
         }
 
+        if (!guard) {
+            guard = co_await start_operation();
+        }
```

Comment:
This if (!guard) guard = looks like a recipe for things working accidentally, no?

We should have clearer transaction boundaries.

Reply:
Even if we always start a new guard here, things may work accidentally in the absence of concurrency, so I don't see that as a reason to avoid the optimization.

This `if (!guard)` line marks the start of transaction. Reuse of the guard is an optimization in case it wasn't consumed by the execution of updates to avoid needless start_operation(), which is relatively expensive as it involves a raft read barrier. In this case, taking the guard is just to humor the `exec_global_command()` API, there is no actual transaction going on.

Reply:
Please add a comment, or encapsulate in a commented start_or_adopt_transaction()